### DISCUSSION TITLE
Absolute stopping criteria normalized by problem size

### DISCUSSION THREAD

**Discussion by blegouix** [ginkgo-project/ginkgo#1570] [Ideas]
Hello,

If I am not wrong, when using:

```
std::shared_ptr const residual_criterion = gko::stop::ResidualNorm<double>::build()
                                                           .with_baseline(gko::stop::mode::absolute)
                                                           .with_reduction_factor(eps)
  ```
  
the norm of residual will be compared with `eps`. However, it means it does not "adapt" to the size of the problem. For example, if I call `gamma` the average residual (typical ||R_ij|| value) for a multiple-rhs problem, the stopping criteria will lead to `norm(R)~m*n_rhs*gamma<eps` and thus `gamma < eps/m/n_rhs`. `gamma` is inversely proportionnal to the size of the problem.

Am I correct ? Is their any way to divide the norm per the size of the problem `m*n_rhs` while using absolute stopping criteria ?

(Note : I may not know `m` and `n_rhs` when building the factory, that's why I cannot juste write `.with_reduction_factor(eps*m*n_rhs)`)

**Comment by MarcelKoch** [comment]
The residual check is applied to each vector of the right-hand-side individually. So we check `||R_:j|| < eps` and not `||R|| < eps`.
